keps/sig-cli/4292-kubectl-debug-custom-profile/README.md

```diff
@@ -0,0 +1,981 @@
+
+- N examples of real-world usage
+- N installs
+- More rigorous forms of testing—e.g., downgrade tests and scalability tests
+- Allowing time for feedback
+
+**Note:** Generally we also wait at least two releases between beta and
+GA/stable, because there's no opportunity for user feedback, or even bug reports,
+in back-to-back releases.
+
+**For non-optional features moving to GA, the graduation criteria must include
+[conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+
+#### Deprecation
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+-->
+
+#### Alpha
+
+- Feature hidden behind an environment variable in kubectl.
+- Unit tests are implemented and enabled.
+
+#### Beta
+
+- Gather feedback from developers and surveys.
+- Environment variable is enabled by default and feature can be disabled explicitly.
+- Integration tests are implemented and enabled.
+
+#### GA
+
+- Environment variable is removed altogether and feature is enabled and can not be disabled.
+- e2e tests are implemented and enabled.
+
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+
+NA
+
+### Version Skew Strategy
+
+<!--
+If applicable, how will the component handle version skew with other
+components? What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- Does this enhancement involve coordinating behavior in the control plane and nodes?
+- How does an n-3 kubelet or kube-proxy without this feature available behave when this feature is used?
+- How does an n-1 kube-controller-manager or kube-scheduler without this feature available behave when this feature is used?
+- Will any other components on the node change? For example, changes to CSI,
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+Copying pod and node debugging use built-in API endpoints. Ephemeral container functionality was
+promoted to stable in 1.25. Besides custom profiling feature only relies on
+`corev1.Container` type's existence differently from `kubectl debug` which is also built-in.
+It is zero probability that this feature touches unavailable API endpoints in regard to
```

Comment:
Nit: I'd stay way from zero probability, but rather use something lighter, like very low probability, low chances, etc. 
1.25 is not that old, and based on the results from past PRR surveys show that a lot of users still keep older versions of k8s. So it's possible that someone might use older than 1.25 k8s, just very small :wink:

Reply:
Indeed you are right. I'll update this statement based on your suggestions. Thanks.